commons-rng-simple/src/main/java/org/apache/commons/rng/simple/internal/ProviderBuilder.java

```diff
@@ -306,9 +306,8 @@ protected Object convertSeed(Object seed) {
                 final long increment = SeedUtils.createLongHexPermutation(source);
                 // The initial state should not be low complexity but the Weyl
                 // state can be any number.
-                final long state = increment;
                 final long weylState = source.nextLong();
-                return new long[] {state, weylState, increment};
+                return new long[] {increment, weylState, increment};
```

Comment:
I agree the code is redundant. But it makes it very clear the returned array is composed of 3 distinct values, the initial state, the state for the Weyl sequence and the increment. This is a micro-optimisation in a method used for a constructor for a long lived object. Performance impact will be zero and code clarity is reduced by the change.